stanza-server/corrector_functions.py

```diff
@@ -322,3 +335,121 @@ def generate_grammar_output(input_text, corrections):
         })
 
     return {"corrector_results": result, "error_list": error_list}
+
+
+def calculate_noun_count(words_types):
+    return sum(1 for word in words_types if word in (NOUN, PROPN))
+
+
+def verb_and_noun_relation(words_types):
+    noun_count = sum(1 for word in words_types if word in (NOUN, PROPN))
+    verb_count = sum(1 for word in words_types if word == VERB)
+
+    if verb_count == 0:
+        return "0.00"
+    return f"{noun_count / verb_count:.2f}"
+
+
+def calculate_abstractness_average(abstract_answer):
+    abstract_count = 0
+    abstractness_sum = 0
+    for word in abstract_answer:
+        if word.get("abstractness") is not None and word.get("pos") == "Nimisõna":
+            abstract_count += 1
+            abstractness_sum += word["abstractness"]
+
+    if abstract_count == 0:
+        return "0.00"
+    return f"{abstractness_sum / abstract_count:.2f}"
+
+
+def calculate_content_word(lemmas, word_types):
+    content_word_count = 0
+    for lemma, pos in zip(lemmas, word_types):
+        if pos != NUM and lemma not in stop_words:
+            content_word_count += 1
+    return content_word_count
+
+
+def calculate_abstract_words(abstract_answer, word_types):
+    abstract_count = 0
+    for index, word in enumerate(abstract_answer):
+        if (word.get("abstractness") == 3 and
+            word_types[index] != PROPN and
```

Comment:
Kui juba muutmiseks läheb, siis imo on parem loetavus kui "and" oleks iga uue rea alguses, mitte eelmise rea lõpus